pkg/strategy/linregmaker/strategy.go

```diff
@@ -0,0 +1,786 @@
+	// allowed.
+	SlowLinReg *indicator.LinReg `json:"slowLinReg"`
+
+	// AllowOppositePosition if true, the creation of opposite position is allowed when both fast and slow LinReg are in
+	// the opposite direction to main trend
+	AllowOppositePosition bool `json:"allowOppositePosition"`
+
+	// FasterDecreaseRatio the quantity of decreasing position orders are multiplied by this ratio when both fast and
+	// slow LinReg are in the opposite direction to main trend
+	FasterDecreaseRatio fixedpoint.Value `json:"fasterDecreaseRatio,omitempty"`
+
+	// NeutralBollinger is the smaller range of the bollinger band
+	// If price is in this band, it usually means the price is oscillating.
+	// If price goes out of this band, we tend to not place sell orders or buy orders
+	NeutralBollinger types.IntervalWindowBandWidth `json:"neutralBollinger"`
+
+	// neutralBoll is the neutral price section for TradeInBand
+	neutralBoll *indicator.BOLL
+
+	// TradeInBand
+	// When this is on, places orders only when the current price is in the bollinger band.
+	TradeInBand bool `json:"tradeInBand"`
+
+	// Spread is the price spread from the middle price.
+	// For ask orders, the ask price is ((bestAsk + bestBid) / 2 * (1.0 + spread))
+	// For bid orders, the bid price is ((bestAsk + bestBid) / 2 * (1.0 - spread))
+	// Spread can be set by percentage or floating number. e.g., 0.1% or 0.001
+	Spread fixedpoint.Value `json:"spread"`
+
+	// BidSpread overrides the spread setting, this spread will be used for the buy order
+	BidSpread fixedpoint.Value `json:"bidSpread,omitempty"`
+
+	// AskSpread overrides the spread setting, this spread will be used for the sell order
+	AskSpread fixedpoint.Value `json:"askSpread,omitempty"`
+
+	// DynamicSpread enables the automatic adjustment to bid and ask spread.
+	// Overrides Spread, BidSpread, and AskSpread
+	DynamicSpread dynamicmetric.DynamicSpread `json:"dynamicSpread,omitempty"`
+
+	// MaxExposurePosition is the maximum position you can hold
+	// 10 means you can hold 10 ETH long/short position by maximum
+	MaxExposurePosition fixedpoint.Value `json:"maxExposurePosition"`
+
+	// DynamicExposure is used to define the exposure position range with the given percentage.
+	// When DynamicExposure is set, your MaxExposurePosition will be calculated dynamically
+	DynamicExposure dynamicmetric.DynamicExposure `json:"dynamicExposure"`
+
+	bbgo.QuantityOrAmount
+
+	// DynamicQuantityIncrease calculates the increase position order quantity dynamically
+	DynamicQuantityIncrease dynamicmetric.DynamicQuantitySet `json:"dynamicQuantityIncrease"`
+
+	// DynamicQuantityDecrease calculates the decrease position order quantity dynamically
+	DynamicQuantityDecrease dynamicmetric.DynamicQuantitySet `json:"dynamicQuantityDecrease"`
+
+	// UseDynamicQuantityAsAmount calculates amount instead of quantity
+	UseDynamicQuantityAsAmount bool `json:"useDynamicQuantityAsAmount"`
+
+	session *bbgo.ExchangeSession
```

Comment:
put private fields to the end of this struct

Reply:
done